R/summarise_country.R

```diff
@@ -0,0 +1,163 @@
+summarise_sheet_indicator_list <- function(billion) {
+  billion <- rlang::arg_match(billion)
+  indicator_order_b <- indicator_order %>%
+    dplyr::filter(ind %in% billion_ind_codes(billion)) %>%
+    dplyr::arrange(order) %>%
+    dplyr::select(sdg, short_name, medium_name, unit_raw, )
+
+  return(indicator_order_b)
+}
+
+#' Summarise Excel sheet 'data' for HPOP country summary
+#'
+#' `summarise_HPOP_cntry_data` summarises the 'data' sgeet for the HPOP country summary.
+#'
+#' @param iso3 ISO3 codes of country to export.
+#' @param ... Additional arguments passed to `calculate_hpop_contributions` and
+#' `calculate_hpop_billion`. Use if you need to provide additional parameters to
+#' those functions.
+#' @inherit calculate_hpop_contributions params
+#'
+#' @return list of data frames to be used in the 'data' sheet of HPOP country summary
+#' export
+#'
+#' @export
+summarise_HPOP_cntry_data <-
+  function(df,
+           iso3,
+           start_year = 2018,
+           end_year = 2019:2023,
+           ...) {
+    assert_mart_columns(df)
+    assert_years(start_year, end_year)
+
+    # Filter df for country, arrange indicators by order.
+    df_iso3 <- df %>%
+      dplyr::filter(iso3 == !!iso3) %>%
+      dplyr::left_join(dplyr::select(indicator_order, ind, order), by = "ind") %>%
```

Comment:
So, in a package environment, you cannot refer to variables directly like this `dplyr::select(indicator_order, ind, order)` unless those have been explicitly passed in the parent function or are defined as global variables (not ideal). To bypass this, I use the method of passing column names as strings. This isn't necessarily optimal, but I think it allows for easier use of multiple columns in some of the transform functions. In this case, since `indicator_order` is a data frame contained within the package, and we know the column names, we can refer to them directly as string values.

`dplyr::select(indicator_order, "ind", "order")`

Now, if the `df` passed into the function may have an `ind` column, but it could have a different name. I've taken the slightly annoying but flexible strategy of allowing different column names. Probably a bad design decision we can reverse, but made sense when this package did a lot less. Because of this, we would pass in the `ind` column name as a string. In this case, `ind = "ind"` in the function definition. And for this line of code, we would then have.

`dplyr::left_join(dplyr::select(indicator_order, "ind", "order"), by = c(ind = "ind"))`

However, one more thing. Since `indicator_order` is an exported dataset from the package, we need to refer to it explicitly as being sourced from the package when calling it, so we would have this as:

`dplyr::left_join(dplyr::select(billionaiRe::indicator_order, "ind", "order"), by = c(ind = "ind"))`

Reply:
That makes a lot of sense, and I will adopt this strategy, even if the declaration of column names is going to be annoying haha.